docs/plugins/components.md

```diff
@@ -0,0 +1,842 @@
+---
+sidebar_position: 5
+title: Components
+description: Documented components for plugins
+---
+
+# Components
+
+:::caution
+
+Any components below whose name ends in `Item` already has a `FormItem` (or equivalent) wrapper. You
+should not use this on those components.
+
+:::
+
+## Input components
+
+All input components ending with `Item` have their original counterpart, and differ in some props:
+
+| Name        | Type                | Default | Description                                      |
+| ----------- | ------------------- | ------- | ------------------------------------------------ |
+| `children?` | React.ReactNode     |         | Wrapper component title                          |
+| `note?`     | string              |         | Description of what the input component is about |
+| `style?`    | React.CSSProperties |         | Wrapper component style                          |
+
+### Checkbox and CheckboxItem
+
+Example:
+
+```tsx
+import { components } from "replugged";
+const { CheckboxItem } = components;
+
+export function Settings(): React.ReactElement {
+  return <CheckboxItem {...util.useSetting(cfg, "foo", "bar")}>Checkbox title</CheckboxItem>;
+}
+```
+
+Properties:
+
+| Property  | Description                                                     |
+| --------- | --------------------------------------------------------------- |
+| `.Aligns` | Checkbox alignments <br/>`TOP` \| `CENTER`                      |
+| `.Shapes` | Checkbox shapes <br/>`BOX` \| `ROUND` \| `SMALL`                |
+| `.Types`  | Checkbox types <br/>`DEFAULT` \| `INVERTED` \| `GHOST` \| `ROW` |
+
+Props:
+
+| Name              | Type                | Default            | Description                                        |
+| ----------------- | ------------------- | ------------------ | -------------------------------------------------- |
+| `align?`          | string              | `center`           | Alignment of the elements inside the checkbox      |
```

Comment:
Instead of the `?` in the name, can you put (optional) in the type?

Reply:
Hmm I think this would look terrible. Almost all props are optional, instead i would put (required) in the type

Reply:
That's fine, just add a note to the top of page that everything is optional unless otherwise specified. Also, `(required)` should be bolded to make sure people don't miss it.